usercp.php

```diff
@@ -3348,6 +3351,68 @@
 	redirect("usercp.php?action=attachments", $lang->attachments_deleted);
 }
 
+
+if($mybb->input['action'] == "warninglog")
+{
+    if(empty($mybb->user['warningpoints']))
+    {
+        error_no_permission();
+    }
+
+	$plugins->run_hooks('usercp_warninglog_start');
+
+	// Pagination
+	$query = "
+		SELECT COUNT(w.wid) AS count
+		FROM " . TABLE_PREFIX . "warnings w
+		LEFT JOIN " . TABLE_PREFIX . "warningtypes t ON w.tid = t.tid
+		WHERE w.uid = {$mybb->user['uid']}
+	";
+	$result = $db->query($query);
+	$total_warnings = (int)$db->fetch_field($result, 'count');
+
+	// Pagination settings
+	$per_page = 10;
+	$page = max(1, $mybb->get_input('page', MyBB::INPUT_INT));
+	$total_pages = max(1, ceil($total_warnings / $per_page));
+	$page = min($page, $total_pages);
+	$start = ($page - 1) * $per_page;
+
+	$url = 'usercp.php?action=warninglog';
+	$multipage = multipage($total_warnings, $per_page, $page, $url);
+
+	$warning_list = [];
+	$query = "
+        SELECT
+            w.wid, w.title as custom_title, w.points, w.dateline, w.issuedby, w.expires, w.expired, w.daterevoked, w.revokedby,
+			w.requiresacknowledgement, w.acknowledged,
+            t.title
+        FROM ".TABLE_PREFIX."warnings w
+            LEFT JOIN ".TABLE_PREFIX."warningtypes t ON (w.tid=t.tid)
+        WHERE w.uid = {$mybb->user['uid']}
+        ORDER BY w.dateline DESC
+        LIMIT {$start}, {$per_page}
+    ";
+
+	$result = $db->query($query);
+	while($row = $db->fetch_array($result))
+	{
+		$row['issued_date'] = my_date('normal', $row['dateline']);
+		$row['revoked_date'] = ($row['daterevoked'] > 0)? my_date('relative', $row['daterevoked']) : $row['daterevoked'];
+		$row['title'] = !empty($row['title']) ? $row['title'] : $row['custom_title'];
+		$row['points'] = ($row['points'] > 0) ? '+' . $row['points'] : $row['points'];
+		$row['expire_date'] = ($row['expired'] === 1) ? $lang->already_expired : (($row['expires'] > 0) ? nice_time($row['expires'] - TIME_NOW) : $lang->never);
```

Comment:
some engines may return strings

```suggestion
		$row['expire_date'] = $row['expired'] ? $lang->already_expired : (($row['expires'] > 0) ? nice_time($row['expires'] - TIME_NOW) : $lang->never);
```